pkg/stub/handler.go

```diff
@@ -574,47 +654,27 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		h.mutex.Lock()
 		defer h.mutex.Unlock()
 
-		err = h.SpecValidation(srcfg)
+		cm, err := h.SpecValidation(srcfg)
 		if err != nil {
 			return err
 		}
 
-		// if the secret event came in before the samples resource event,
-		// it could not be processed (though it would have been cached in
-		// the Handler struct);  process it now
-		if h.registrySecret != nil &&
-			!srcfg.ConditionTrue(v1alpha1.ImportCredentialsExist) {
-			h.manageDockerCfgSecret(false, srcfg, h.registrySecret)
+		if cm != nil {
+			changed, err := h.VariableConfigChanged(srcfg, cm)
+			if err != nil {
+				return err
+			}
+			if !changed {
+				return nil
+			}
 		}
 
 		// if trying to do rhel to the default registry.redhat.io registry requires the secret
 		// be in place since registry.redhat.io requires auth to pull; if it is not ready
-		// log error state
-		if srcfg.Spec.InstallType == v1alpha1.RHELSamplesDistribution &&
-			(srcfg.Spec.SamplesRegistry == "" || srcfg.Spec.SamplesRegistry == "registry.redhat.io") &&
-			!srcfg.ConditionTrue(v1alpha1.ImportCredentialsExist) &&
-			h.registrySecret == nil {
-			if h.waitingForCredential == nil {
-				h.waitingForCredential = srcfg
-				err := fmt.Errorf("Cannot create rhel imagestreams to registry.redhat.io without the credentials being available: %#v", srcfg)
-				return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionFalse, err, "%v")
-			} else {
-				// already recorded error/need, simply wait for credential to appear, but we'll update our cached
-				// entry for resource version validation on various status condition updates.  That said, the basic
-				// flow will be
-				// 1) cache in waitingForCredential any updates to samplesresoure
-				// 2) get secret event
-				// 3) secret event is processed, samplesresource condition is updated with credentials exists, using cwaitingForCredential
-				// 4) new samplesresouce event comes in based on status update from 3)
-				// 5) it is processed since credential exits
-				// 6) if successful the samplesexists condition is created, and the configmap is populated, and samplesResource cache is updated
-				h.waitingForCredential = srcfg
-				return nil
-			}
-		} else {
-			// if we change config in the interim to no longer require creds, nil out waitingForCredentials
-			// so it does not impact any future checks
-			h.waitingForCredential = nil
+		// error state will be logged by WaitingForCredential, so return error and requeue
+		err = h.WaitingForCredential(srcfg)
+		if err != nil {
+			return err
```

Comment:
I don't think we need to requeue here.

When we see the user's secret, we should the openshift secret and update the samplesresource.status.conditions to include CredentialsExist=true

The handler should then see an update event for SamplesResource(the one that updated the status), recognize the credentials exist, and proceed to create the samples.

Reply:
not returning error / not initiating requeue part of next push